app/scripts/controllers/permissions.js

```diff
@@ -207,6 +238,32 @@ class PermissionsController {
             return end()
           },
         },
+        'eth_addPlugin_*': {
+          description: 'Install plugin $1, which will download new functionality to MetaMask.',
+          method: async (req, res, next, end) => {
+            const pluginNameMatch = req.method.match(/eth_addPlugin_(.+)/)
+            const pluginName = pluginNameMatch && pluginNameMatch[1]
+            const sourceUrl = req.params[0].sourceUrl
+
+            const response = await this.pluginsController.add(pluginName, sourceUrl)
+            return response
+          },
+        },
+        'eth_runPlugin_*': {
```

Comment:
So the way I had built rpc-cap's wildcard support, I had built these to basically be one, ie maybe we'd just have `wallet_connectToPlugin_*`, which would install if it wasn't available, and otherwise would request a connection to that plugin, which then has the ability to define its own permissions internally.

Anyways, we can change how it's patterned, but that's the usage that would work with rpc-cap as it is today.